### PR TITLE
Avoid object[] alloc when logging

### PR DESCRIPTION
**Pull request by davkean**
Calls to FormatResourceString were allocating object[] - replace with calls to GetResourceString.

### CODE REVIEW

File: src/Shared/ResourceUtilities.cs [hunk 1]
```diff
@@ -171,6 +172,13 @@ internal static string FormatResourceString(out string code, out string helpKeyw
             return ExtractMessageCode(true /* msbuildCodeOnly */, FormatString(GetResourceString(resourceName), args), out code);
         }
 
+        [Obsolete("Use GetResourceString instead.", true)]
```
Comment (davkean) [inline src/Shared/ResourceUtilities.cs]: Note, new code start here.

**Comment by davkean** [comment]
tag @pilchie

File: src/Shared/ResourceUtilities.cs (hunk 1 above)
Comment (Pilchie) [reply src/Shared/ResourceUtilities.cs]: Wait, how does this work?  Shouldn't this method at least have a single `string` parameter?

Comment (davkean) [reply src/Shared/ResourceUtilities.cs]: Whoops, I must have removed the string parameter. I used this method to find the original usages so looks like I must have removed it when I added the obsolete.

**Pull request merged**